packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts

```diff
@@ -0,0 +1,63 @@
+import { StateUpdateListener } from '../interface';
+import { AppMetadata, RequestArguments, Signer } from ':core/provider/interface';
+import { AddressString } from ':core/type';
+import { CBInjectedProvider, CBWindow } from ':util/provider';
+
+export class ExtensionSigner implements Signer {
+  private readonly metadata: AppMetadata;
+  private readonly updateListener: StateUpdateListener;
+  private extensionProvider: CBInjectedProvider | undefined;
+
+  constructor(params: { metadata: AppMetadata; updateListener: StateUpdateListener }) {
+    this.metadata = params.metadata;
+    this.updateListener = params.updateListener;
+    const window = globalThis as CBWindow;
```

Comment:
can we do it without introducing CBWindow

Reply:
How's this? [ef62054](https://github.com/coinbase/coinbase-wallet-sdk/pull/1332/commits/ef620542631a2df0ed4b812d3fc8a360b5502cfd)